### ISSUE TITLE
(At least) the sorting notebook is erroneous

### ISSUE THREAD

**Issue by hoefling** [higgsfield/np-hard-deep-reinforcement-learning#4]
Brief looking at the code:
```python
    for i in range(seq_len):
        ...
        for i in range(self.n_glimpses):
            ...
        loss += self.criterion(logits, target[:,i])
```
`i` will always be `self.n_glimpses - 1`, so the displayed loss is meaningless. Sorting predicted by the model will also be meaningless. After fixing it, the loss won't change and the model doesn't train at all. Logits are not softmaxed after pointing. Masks are never applied since `idxs` are never initialized.
